freedomofpress/securedrop
# Automate updating screenshots in user guides with Selenium page layout tests

## Status

Ready for review

## Description of Changes

This PR: 

* Automates the screenshots in each user guide (closes #2016) by using the Selenium page layout test PNG output files (and adds new page layout tests if the right output PNG did not exist)
* Adds a Makefile target `make update-user-guides` to autoupdate all the screenshots in the user guides (runs the page layout tests and then copies all the PNGs into the right place under `docs/`)
* Adds a note in the developer documentations describing how to update screenshots in the user guides

## Testing

0. Look at the user guides as is in the PR. Some images I have modified to make them easier to maintain. 
1. Run `make update-user-guides` _not_ in a development VM and see it print "You must run this from the development VM!".
2. Run `make update-user-guides` in a development VM and see the page layout tests run and the output be copied to the documentation directory.

## Deployment

This touches only tests and documentation. 

## Checklist

### If you made changes to documentation:
